File: src/sql/postgres/types/bytea.zig
pub const to = 17;
pub const from = [_]short{17};

pub fn toJS(
    globalObject: *jsc.JSGlobalObject,
    value: *Data,
) AnyPostgresError!JSValue {
    defer value.deinit();

    // var slice = value.slice()[@min(1, value.len)..];
    // _ = slice;
    return JSValue.createBuffer(globalObject, value.slice(), null);
}

const bun = @import("bun");
const AnyPostgresError = @import("../AnyPostgresError.zig").AnyPostgresError;
const Data = @import("../../shared/Data.zig").Data;

const int_types = @import("./int_types.zig");
const short = int_types.short;

const jsc = bun.jsc;
const JSValue = jsc.JSValue;

File: src/sql_jsc/postgres/types/bytea.rs
use crate::jsc::{ArrayBuffer, JSGlobalObject, JSValue, js_error_to_postgres};
use bun_sql::postgres::AnyPostgresError;
use bun_sql::shared::Data;

// PostgresString.rs.
pub trait ByteaToJs {
    fn bytea_to_js(self, global: &JSGlobalObject) -> Result<JSValue, AnyPostgresError>;
}

// Takes `Data` by value; Drop at scope exit frees the decode buffer.
impl ByteaToJs for Data {
    fn bytea_to_js(self, global: &JSGlobalObject) -> Result<JSValue, AnyPostgresError> {
        // var slice = value.slice()[@min(1, value.len)..];
        // _ = slice;
        //
        // Use the copying Buffer constructor: `self.slice()`
        // borrows a transient decode buffer that `Drop` frees on return, so
        // JSC must own its own copy.
        ArrayBuffer::create_buffer(global, self.slice()).map_err(js_error_to_postgres)
    }
}
